docs/developer_guide/conversion.md

```diff
@@ -4,4 +4,81 @@ title: Model Conversion Reference
 
 !!! warning
 
-    Looking to implement your own converter? This reference is on the way. Stay tuned!
+    This reference is a work in progress. Stay tuned for new updates!
+
+This reference guide describes all there is to know about Fast-LLM's checkpoint conversion system.
+After reading this, you should be able to create your own `External` converter, in Hugging Face format or other.
+And if you are familiar with the rest of Fast-LLM, you will also be able to create an entirely custom converter.
+
+## Conversion and checkpointing basics
+
+Fast-LLM provides a simple and [fully customizable interface](https://github.com/ServiceNow/Fast-LLM/blob/main/fast_llm/engine/checkpoint/config.py) to save/load checkpoints and configurations.
+This same interface is used both by Fast-LLM official checkpoint formats (`distributed` and `fast-llm`),
+and by the checkpoint conversion interface.
+It can also be used to define entirely new checkpoint formats, though this is generally not recommended.
+
+In this guide we focus on the checkpoint conversion interface, in particular for Hugging Face formats,
+since this is the most common use case.
+
+## Checkpoint format metadata
+
+When creating a new checkpoint format, the first step is to subclass `CheckpointFormat`.
+This data structure holds important properties of the format, and makes them accessible at the configuration level.
+Some important entries include:
+
+*   `name`: A name for the format, as will appear for example in configuration files
+*   `support_optimizer`: Whether the optimizer state can be included in a checkpoint.
+*   `support_saving`, `support_loading`: This can be used to create read-only or write-only formats.
+*   `get_handler_class()`: Return the actual checkpoint conversion class, as we'll soon describe.
+The class should be imported lazily so the `CheckpointFormat` remains accessible by configurations.
+
+Here is a simple example:
+
+```python
+class AwesomeCheckpointFormat(CheckpointFormat):
+    name = "awesome_checkpoint"
+    support_optimizer = False
+
+    @classmethod
+    def get_handler_class(cls):
+        from package.module import AwesomeCheckpointHandler
+
+        return AwesomeCheckpointHandler
+```
```

Comment:
this confuses me a little.
is an example for a checkpoint format "huggingface"? or is it "llama"?
because of that, I think it would be good if you used here a more concrete example, say "LlamaCheckpointFormat" (if that was appropriate).

Reply:
I'm thinking of having a slightly more abstract example so we have more freedom to demonstrate exactly the features we want to. The idea would be to have some kind of custom `AwesomeModel` we follow over the whole guide.